modules/RsPaymentTerminal/src/main.rs

```diff
@@ -188,26 +205,43 @@ impl PaymentTerminalModule {
         };
         let card_info = read_card_loop()?;
 
-        let provided_token = match card_info {
+        if let Some(provided_token) = match card_info {
             CardInfo::Bank => {
-                self.feig.begin_transaction(&token)?;
-
-                // Reuse the bank token as invoice token so we can use the
-                // invoice token later on to commit our transactions.
-                ProvidedIdToken::new(token, AuthorizationType::BankCard)
-            }
-            CardInfo::MembershipCard(id_token) => {
-                ProvidedIdToken::new(id_token, AuthorizationType::RFID)
+                if !self.accept_credit_cards {
+                    None
+                } else {
+                    self.feig.begin_transaction(&token)?;
+                    let credit_cards_connectors = if self.credit_cards_connectors.is_empty() {
+                        None
```

Comment:
this is not what we've documented. in this case we want to publish the token open for all connectors